source/Sashimi/CertificateEncoder.cs

```diff
@@ -150,14 +150,14 @@ static void GrantCurrentUserAccessToPrivateKeyDirectory(string privateKeyPath)
             var security = directory.GetAccessControl();
             security.AddAccessRule(new FileSystemAccessRule(current.User, FileSystemRights.FullControl, InheritanceFlags.ContainerInherit | InheritanceFlags.ObjectInherit, PropagationFlags.None, AccessControlType.Allow));
             directory.SetAccessControl(security);
-#pragma warning restore PC001
+#pragma warning restore CA1416
         }
 
         static bool HasPrivateKey(X509Certificate2 certificate2)
         {
             try
             {
-                return certificate2.HasPrivateKey && certificate2.PrivateKey != null;
+                return certificate2.HasPrivateKey && certificate2.GetRSAPrivateKey() != null;
```

Comment:
.PrivateKey has been deprecated in favour of [GetRSAPrivateKey()](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.rsacertificateextensions.getrsaprivatekey?view=net-6.0) and both return a nullable if the cert doesn't have a valid private key. Given the [exceptions on .PrivateKey](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificate2.privatekey?view=net-6.0#System_Security_Cryptography_X509Certificates_X509Certificate2_PrivateKey) i.e. the key has to be an RSA or DSA key I think it's safe to just check for these two private key types. 

![image](https://user-images.githubusercontent.com/101079287/184050353-fac3ecfb-a5b3-4f21-807e-46e84e3fb732.png)

Reply:
I was missing the DSA key type previously and just added that. Keen to get your thoughts on this.

Reply:
Yeah, I don't know 100% what needs to be supported here, but it feels that if we're migrating from the deprecated property that used to support both, we should check both. As long as those extension methods don't throw exceptions if the Private Key doesn't exist...?